specs/DOMContentLoaded.md

```diff
@@ -0,0 +1,119 @@
+```
+ScenarioDOMContentLoaded::ScenarioDOMContentLoaded(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    //! [DOMContentLoaded]
+    // Register a handler for the DOMContentLoaded event.
+    // Event is raised when the DOM content is loaded
+    CHECK_FAILURE(m_webView->add_DOMContentLoaded(
+        Callback<ICoreWebView2DOMContentLoadedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2DOMContentLoadedEventArgs* args)
+                -> HRESULT {
+                m_webView->ExecuteScript(
+                    LR"~(
+                    let content=document.createElement("h2");
+                    content.style.color='blue';
+                    content.textContent="This text was added by the host app";
+                    document.body.appendChild(content);
+                    )~",
+                    Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                        [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                        .Get());
+                return S_OK;
+            })
+            .Get(),
+        &m_DOMContentLoadedToken));
+    //! [DOMContentLoaded]
+```
+
+## C#
+```
+webView.CoreWebView2.DOMContentLoaded += (object sender, CoreWebView2DOMContentLoadedEventArgs arg) =>
+{
+    webView.ExecuteScriptAsync("let " +
+                              "content=document.createElement(\"h2\");content.style.color=" +
+                              "'blue';content.textContent= \"This text was added by the " +
+                              "host app\";document.body.appendChild(content);");
+};
+webView.NavigateToString(@"<!DOCTYPE html><h1>DOMContentLoaded sample page</h1><h2>The content below will be added after DOM content is loaded </h2>");
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2_2;
+interface ICoreWebView2DOMContentLoadedEventArgs;
+interface ICoreWebView2DOMContentLoadedEventHandler;
+
+[uuid(9810c82b-8483-4f1c-b2f4-6244f1010c05), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
+  /// Add an event handler for the DOMContentLoaded event.
+  /// DOMContentLoaded is raised when the initial html document has been parsed.
+  /// This aligns with the the document's DOMContentLoaded event in html
+  ///
+  /// \snippet ScenarioDOMContentLoaded-Staging.cpp
+  HRESULT add_DOMContentLoaded(
+      [in] ICoreWebView2StagingDOMContentLoadedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_DOMContentLoaded.
+  HRESULT remove_DOMContentLoaded(
+      [in] EventRegistrationToken token);
+}
+
+/// Event args for the DOMContentLoaded event.
+[uuid(E8BA4206-D6F8-42F1-9A6D-43C8A99C1F39), object, pointer_default(unique)]
+interface ICoreWebView2DOMContentLoadedEventArgs : IUnknown {
+  /// The ID of the navigation.
```

Comment:
Is this emitted from some prior NavigateTo method? How should the app use or manage this ID?

Reply:
It corresponds to other navigation ID properties on other navigation events. Please update the docs for the NavigationId property to note this.